### ISSUE TITLE
Pre-training stage1 BLIP2 or your work

### ISSUE THREAD

**Issue by jaeseokbyun** [yiren-jian/BLIText#2]
First of all, thanks for sharing great work!

I try reproduce your work and BLIP2 for 4M dataset in 8 A100 gpus (40GB)
But, It takes 15 hours for a single epoch in stage1 for both BLIP2 (LAVIS) and your work.

Could you give some advice for accelerating this? In your paper, only takes 66 hours for whole 10 epochs.

**Comment by yiren-jian** [comment]
I trained BLIP2 on eight RTX-A6000 and eight A100-80G. Training time is manageable. I can only provide some guesses in resolving your issues:

-  image size: storing images using 256x256 or 384 x 384 on disk. Large size image will make IO an issue.
- Using SSD over HD to store your dataset. 
- Make sure you freeze ViT and not using gradient checkpointing, through the configuration files. You can also check my configuration and training log [here](https://github.com/yiren-jian/BLIText/blob/main/training_logs/BLIP2/training_stage1.txt) (trained on 8x A6000). Each epoch or mine is about 4.5 hrs.
- I fit 128 images in a single RTX-A6000 GPU (total batch size 8x 128), you should have a similar number in A100-40G. My empirical study shows that A100 can fit more images than A6000, in the same amount of VRAM.

I have not really done anything special in adapting LAVIS code. I also strictly follow the pytorch version suggested.

**Comment by yiren-jian** [comment]
If you're running 2 servers with 4 GPUs each, communication could also be a bottleneck. As baseline, training with 4 GPUs could also give OK models.

**Comment by jaeseokbyun** [comment]
Thanks for fast reply! I think the main reason would be the image size. I used the original size images on disk. 

Moreover, I also wonder that your CC3M and SBU captions are constructed from BLIP 129M? or 14M?

**Comment by yiren-jian** [comment]
No, for this paper, the captions for CC3M and SBU are original. I follow [this](https://github.com/salesforce/LAVIS/tree/main/lavis/datasets/download_scripts).

**Comment by jaeseokbyun** [comment]
Thanks! Actually, when we tried ALBEF, and BLIP models to reproduce (not LAVIS), their training speed is decent for our setup. (which uses the same original sized dataset). Therefore, I'm afraid the image size would not be the main reason of slow BLIP2

I think I need to check the software version

**Comment by jaeseokbyun** [comment]
Could you share the software version ? (CUDA, pytorch) ?

**Comment by yiren-jian** [comment]
CUDA 11, and I tried both pytorch 1.8 and pytorch 2.0

**Comment by jaeseokbyun** [comment]
Thanks for your help!
